tests/apidoc_login_test.go

```diff
@@ -0,0 +1,41 @@
+package tests
+
+import (
+	"encoding/json"
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+	"github.com/tidwall/gjson"
+	"testing"
+)
+
+func TestLogin(t *testing.T) {
+	deployment := Deploy(t, "login", b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	unauthedClient := deployment.Client(t, "hs1", "")
+	t.Run("parallel", func(t *testing.T) {
+		t.Run("POST /login can login as user", func(t *testing.T) {
+			t.Parallel()
+			res := unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "register"}, json.RawMessage(`{
+				"auth": {
+					"type": "m.login.password"
+				},
+				"username": "post-login-user",
+				"password": "superuser"
+			}`))
+			res = unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "login"}, json.RawMessage(`{
+				"auth": {
+					"type": "m.login.dummy"
+				},
+				"username": "post-login-user",
+				"password": "superuser"
+			}`))
+			must.MatchResponse(t, res, match.HTTPResponse{
+				JSON: []match.JSON{
+					match.JSONKeyTypeEqual("access_token", gjson.String),
+				},
```

Comment:
It's really important that the Complement tests we make are as thorough as the Sytests we're replacing. The login test we're replacing also checks the `home_server` - https://github.com/matrix-org/sytest/blob/develop/tests/10apidoc/02login.pl#L96 so please do the same. The homeserver should `== "hs1"`.

Reply:
Hi @Kegsay , I have made the change